### ISSUE TITLE
Duplicate ID

### ISSUE THREAD

**Issue by dcalixto** [formtastic/formtastic#1192]
On a user edit page i have several forms using formstatic but formstatic is duplicating the ids, i would like to know if have a way to change this issue. here a exemple:

`<form accept-charset="UTF-8" action="/users/test" class="formtastic user" id="edit_user_1" method="post" novalidate="novalidate">`

the edit_user_1 is duplicating on the others forms.

thank's

**Comment by mikz** [comment]
See the "Namespaced forms all multiple Formtastic forms to exist on the one page without DOM id clashes and invalid HTML documents." in the [documentation of `semantic_form_for`](http://www.rubydoc.info/github/justinfrench/formtastic/Formtastic/Helpers/FormHelper#semantic_form_for-instance_method). 

That should solve your problem.

**Comment by dcalixto** [comment]
@mikz  thank's for your reply, i placed the namespace  and did not solve.

**Comment by mikz** [comment]
@dcalixto you can still pass `:html` option to it. So something like this would solve your problem:

```ruby
semantic_form_for(@user, html: { id: 'prefixed_form' })
```

**Comment by dcalixto** [comment]
thank you very much @mikz , and unintentionally abuse, the same probem seems to appear on submit:

i've tried the same logic but seems not solve.

Duplicate ID "user_submit_action"

```
  <%= form.actions do %>
        <%= form.action(:submit, :label => "Save",  :button_html => { :class => "btn", id: 'button_address'}) %>
        <% end %>
```

**Comment by mikz** [comment]
@dcalixto could you post the generated html ? It is weird, that `button_html` would not work.

**Comment by mikz** [comment]
@dcalixto I check the source and the tests and I don't see any possibility why `:namespace` option should not work. Could you paste an example how you were trying to use it?

It would help us a lot. To improve documentation, we need to learn how people use it and what mistakes they make.

**Comment by fredngo** [comment]
I too am experiencing the same problem with the `namespace` option. The versions used are: formtastic (3.1.4) and formtastic-bootstrap (3.1.1). (I wonder if it's a formtastic-bootstrap problem.)

Using the plain old Rails form builder:

```ruby
= form_for(@animal, namespace: 'cat', url: zoo_animal_path) do |f|
```

gets me:

```html
<form id="cat_edit_zoo_animal_249" class="edit_zoo_animal" method="post" accept-charset="UTF-8" action="/zoo/animals/249">
```

whereas using Formtastic:

```ruby
= semantic_form_for(@animal, namespace: 'cat', url: zoo_animal_path) do |f|
```

gets me:

```html
<form id="edit_zoo_animal_249" class="formtastic zoo_animal" method="post" accept-charset="UTF-8" action="/zoo/animals/249" novalidate="novalidate">
```

So the `cat` namespace is missing in the `id`.

**Comment by mikz** [comment]
@fredngo Thanks for the report! I made a PR (#1212) with a regression test to verify it surely fails. And it does. 

I'm not sure why, how to fix it or how much time I'll have to work on that. If you feel like, feel free to chip in.

**Comment by fredngo** [comment]
Wow, thanks for the quick response @mikz ! For now that the issue is documented at least.

**Comment by mateusg** [comment]
I'm experiencing the same problem with duplicate ids on the submit buttons and forms.

I think this is happening because the `:namespace` option isn't being passed to the [original implementation](http://api.rubyonrails.org/classes/ActionView/Helpers/FormHelper.html#method-i-form_for) of `form_for`. It [gets remapped](https://github.com/justinfrench/formtastic/blob/master/lib/formtastic/helpers/form_helper.rb#L158-L174) as `custom_namespace`.

@mikz Do you know if there's a reason it was remapped like this?

**Comment by mikz** [comment]
I made https://github.com/justinfrench/formtastic/pull/1065to to fix https://github.com/justinfrench/formtastic/issues/904. It was doing nasty stuff and changing class variables when rendering forms.

It is possible I did not fully understand the implications and it has a bug that prevents using the `:namespace` option. We should verify the regression test https://github.com/justinfrench/formtastic/pull/1212 failed/passed before these changes. If so then it indeed broke it.
